river/Seat.zig

```diff
@@ -164,6 +164,10 @@ pub fn focus(self: *Self, _target: ?*View) void {
             unreachable;
         }
         self.setFocusRaw(.{ .view = view });
+        if (view.current.urgent) {
+            view.current.urgent = false;
+            view.output.sendUrgentTags();
+        }
```

Comment:
In order to ensure frame perfection, we can't modify the current (i.e. displayed) state here. Instead we should set the pending state and call sendUrgentTags() in `Root.commitTransaction()` or `XdgToplevel.handleCommit()` if the pending state being applied differs from the current state.

Also this check should be moved to `setFocusRaw()` where `view.setActivated(false)` is called.

Reply:
Same question as above, do I need a new loop in `Root.commitTransaction()` and `XdgToplevel.handleCommit()` or can I call `output.sendUrgentTags();` and `view.output.sendUrgentTags();`?

Reply:
In `Root.commitTransaction()` and `XdgToplevel.handleCommit()` we need to send new information to all river status clients if the tags containing urgent views have changed. It looks like your `Output.sendUrgentTags()` function sends the current set of urgent tags on the output to all clients, so it would make sense to use that.

I believe there are three cases in which the set of urgent tags for a given output can change:

1. the urgent state of some view on the output changes
2. the tags of an urgent view change
3. an urgent view is sent to a different output (need o send new set of urgent tags for both outputs)

In each of these cases, we must send the new urgent tag set to all river-status clients.